src/app/services/share.js

```diff
@@ -132,6 +122,115 @@ module.exports = (Model, App) => {
     return newShare.token;
   };
 
+  /**
+   * Generates a share folder token
+   * @param user
+   * @param folderId
+   * @param bucket
+   * @param mnemonic
+   * @param fileToken
+   * @param views
+   * @returns {Promise<string|*>}
+   * @constructor
+   */
+  const GenerateFolderTokenAndCode = async (
+    user,
+    folderId,
+    bucket,
+    mnemonic,
+    fileToken,
+    views = 1,
+  ) => {
+    const itemExists = await Model.folder.findOne({
+      where: {
+        id: { [Op.eq]: folderId },
+        user_id: { [Op.eq]: user.id },
+      }
+    });
+
+    if (!itemExists) {
+      throw Error('Folder not found');
+    }
+
+    const maxAcceptableSize = 1024 * 1024 * 1000; // 1GB
+
+    const folderSize = await getFolderSize(folderId);
+
+    if (folderSize > maxAcceptableSize) {
+      throw Error('Folder too large');
+    }
+
+    // Generate password
+    const code = crypto.randomBytes(32).toString('hex');
```

Comment:
Private things like the `code` should be generated on the client-side ('crypto' is available there too), same with the encryption, to provide a real zero-knowledge experience. 

Then, we never-ever can get access nor the code or the mnemonic, we just receive the mnemonic already encrypted.